src/infill/ZigzagConnectorProcessorDisconnectedEndPieces.cpp

```diff
@@ -19,54 +19,64 @@ void ZigzagConnectorProcessorDisconnectedEndPieces::registerScanlineSegmentInter
     }
     else
     {
-        if (previous_scanline_is_even && !this_scanline_is_even)
-        { // if we left from an even scanline, but not if this is the line segment connecting that zigzag_connector to an even scanline
-            addLine(last_connector_point, intersection);
-        }
-        else if (!previous_scanline_is_even && !this_scanline_is_even) // if we end an odd boundary in an odd segment
-        { // add whole oddBoundarySegment (including the just obtained point)
-            for (unsigned int point_idx = 1; point_idx < zigzag_connector.size(); point_idx++)
+        const bool is_end_piece = previous_scanline_is_even == this_scanline_is_even;
+        const bool add_connection = previous_scanline_is_even || is_end_piece;  // add connections on even segments or it is an endpiece
+
+        if (add_connection)
```

Comment:
I always find it easier to read when it's like this:

    if (<condition>)
    {
        <do stuff>
    }

Rather than:

    const bool should_do_stuff = <condition>;
    if (should_do_stuff)
    {
        <do stuff>
    }

It's sort of what if-clauses are made for.